cpp/src/arrow/io/memory-benchmark.cc

```diff
@@ -15,50 +15,96 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include <immintrin.h>
```

Comment:
Well, this is almost certainly going to fail on Windows. Since we aren't building the benchmarks in CI on Windows I guess we can let this slide for now. cc @bkietz